packages/resources/src/RemixSite.ts

```diff
@@ -0,0 +1,1506 @@
+        zone: this.cdk.hostedZone,
+        recordNames: [domainAlias],
+        targetDomain: recordName,
+      });
+    }
+  }
+
+  /////////////////////
+  // Helper Functions
+  /////////////////////
+
+  private registerSiteEnvironment() {
+    const environmentOutputs: Record<string, string> = {};
+    for (const [key, value] of Object.entries(this.props.environment || {})) {
+      const outputId = `SstSiteEnv_${key}`;
+      const output = new CfnOutput(this, outputId, { value });
+      environmentOutputs[key] = Stack.of(this).getLogicalId(output);
+    }
+
+    const root = this.node.root as App;
+    root.registerSiteEnvironment({
+      id: this.node.id,
+      path: this.props.path,
+      stack: Stack.of(this).node.id,
+      environmentOutputs,
+    } as BaseSiteEnvironmentOutputsInfo);
+  }
+
+  private readRemixConfig(): RemixConfig {
+    const { path: sitePath } = this.props;
+
+    const configDefaults: RemixConfig = {
+      assetsBuildDirectory: "public/build",
+      publicPath: "/build/",
+      serverBuildPath: "build/index.js",
+      serverBuildTarget: "node-cjs",
+      server: undefined,
+    };
+
+    // Validate config path
+    const configPath = path.resolve(sitePath, "remix.config.js");
+    if (!fs.existsSync(configPath)) {
+      throw new Error(
+        `Could not find "remix.config.js" at expected path "${configPath}".`
+      );
+    }
+
+    // Load config
+    const userConfig = require(configPath);
+    const config: RemixConfig = {
+      ...configDefaults,
+      ...userConfig,
+    };
+
+    // Validate config
+    Object.keys(configDefaults)
+      .filter((key) => key !== "server")
+      .forEach((key) => {
+        const k = key as keyof RemixConfig;
+        if (config[k] !== configDefaults[k]) {
+          throw new Error(`RemixSite: remix.config.js "${key}" must be "${configDefaults[k]}".`);
+        }
+    });
+
+    return config;
+  }
+
+  private generateBuildId(): string {
+    // We will generate a hash based on the contents of the "public" folder
+    // which will be used to indicate if we need to invalidate our CloudFront
+    // cache. As the browser build files are always uniquely hash in their
+    // filenames according to their content we can ignore the browser build
+    // files.
+
+    // The below options are needed to support following symlinks when building zip files:
+    // - nodir: This will prevent symlinks themselves from being copied into the zip.
+    // - follow: This will follow symlinks and copy the files within.
+    const globOptions = {
+      dot: true,
+      nodir: true,
+      follow: true,
+      ignore: ["build/**"],
+      cwd: path.resolve(this.props.path, "public"),
+    };
+    const files = glob.sync("**", globOptions);
+    const hash = crypto.createHash("sha1");
+    for (const file of files) {
+      hash.update(file);
+    }
+    const buildId = hash.digest("hex");
+
+    logger.debug(`Generated build ID ${buildId}`);
+
+    return buildId;
```

Comment:
**praise:** Nice one! This means we can remove the `folder-hash` dependency too.